atmo/clusters/views.py

```diff
@@ -25,10 +31,10 @@ def new_cluster(request):
             )
         )
         return redirect('keys-new')
-    initial = {
-        'identifier': '{}-telemetry-analysis'.format(user_display(request.user)),
-        'size': 1,
-    }
+    elif ssh_key_count == 1:
+        # If only 1 ssh key, make it pre-selected.
+        initial['ssh_key'] = request.user.created_sshkeys.values('pk')[0]['pk']
```

Comment:
So I'm a fan of using the new queryset APIs, e.g. `first()` since it returns None if there is none. That way you can combine the multiple calls from line 22 to here.

E.g.:

```python
first_ssh_key = request.user.created_sshkeys.first()  # order by date?

if first_ssh_key is None:
    # message and redirect to new key view
else:
    initial['ssh_key'] = first_ssh_key.pk
```

That'll even work if there are multiple keys available and select the first one. In case you're doing ordering on the base query set, make sure to update the queryset of the `ModelChoiceField` in `cluster/forms.py`